### ISSUE TITLE
Bug in dropdown view leads to wrong position of dropdown when no optimal position is found

### ISSUE THREAD

**Issue by nilsfr** [ckeditor/ckeditor5#17220]
[The line with the error](https://github.com/ckeditor/ckeditor5/blob/683a4cf3a0500006f21afdbc86c371f22e19153c/packages/ckeditor5-ui/src/dropdown/dropdownview.ts#L289) is the following:
```
optimalPanelPosition ? optimalPanelPosition.name : this._panelPositions[ 0 ].name
```
The problem here is that `this.panelPositions[0]` returns a callable. Here you get the name of the callable, not the name of the position.

The values returned by `_panelPositions` is defined i [defaultPanelPositions](https://github.com/ckeditor/ckeditor5/blob/683a4cf3a0500006f21afdbc86c371f22e19153c/packages/ckeditor5-ui/src/dropdown/dropdownview.ts#L469).

**Comment by nilsfr** [comment]
I tried to fix #17225 by setting `viewportTopOffset`on StickyPanelView to a high number to avoid the panel becoming sticky. The result of this was that `optimalPanelPosition` was not found and the fallback `this._panelPositions[ 0 ].name` was used. But as this fallback gave the name of the function `southEast ` and not the name of the name of the position `se` the classname on the dropdown was wrong, leading to the wrong position of the dropdown.

I found another solution to this so I do not have a screenshot of this, but the code here is wrong as it give the name of the function and not the name of the position.
